### PR TITLE
fix mocks in native storybook

### PR DESCRIPTION
**Pull request by jakob223**
This only works on iOS, I need help making it work on android.

cc @keybase/y2ksquad

### CODE REVIEW

File: shared/ios/Keybase/AppDelegate.m [hunk 1]
```diff
@@ -102,7 +103,9 @@ - (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
 #if DEBUG
   // uncomment to get a prod bundle. If you set this it remembers so set it back and re-run to reset it!
 //  [[RCTBundleURLProvider sharedSettings] setEnableDev: false];
-  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
+  
+  NSString *bundlerURL = IS_STORYBOOK ? @"storybook-index" : @"normal-index";
```
Comment (chrisnojima) [inline shared/ios/Keybase/AppDelegate.m]: maybe we should put a comment here explaining whats going on

File: shared/metro.config.js [hunk 2]
```diff
@@ -7,13 +7,48 @@
...
+      newModuleName = './index'
+      console.log('Switching to storybook mode')
+    } else if (moduleName === './normal-index') {
+      storybook = false
+      newModuleName = './index'
+      console.log('Switching to normal mode')
+    }
+    if (storybook) {
+      newModuleName = newModuleName
+        .replace(/typed-connect/, '__mocks__/typed-connect')
```
[the remark this answers is not in the document]
Comment (jakob223) [reply shared/metro.config.js]: One of the three places is in `package.json` which seems hard to pull in from a random JS file. I added comments to all three places though.

[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/metro.config.js]: we could put this into jest.config.js instead, likely better

**Comment by jakob223** [comment]
- [ ] after code freeze

**Comment by jakob223** [comment]
Bump @chrisnojima

**Comment by jakob223** [comment]
Not sure why I thought this was in your court @chrisnojima , gonna move stuff to one file.

**Comment by chrisnojima** [comment]
after this passes ci can you validate (again) it works correctly with
1. hot server
1. packaged build
1. ios
1. android
1. jest test pass

**Comment by jakob223** [comment]
I can't do android (don't have a working environment but I'll try to set one up again today). How does one do a packaged build?

**Comment by jakob223** [comment]
- [ ] hot server
 - [ ] packaged build
 - [ ] storybook
 - [ ] regular on iOS
 - [ ] storybook on iOS
 - [ ] regular on android
 - [ ] storybook on android
 - [ ] jest

**Comment by jakob223** [comment]
Okay @chrisnojima I have verified that it works on all these platforms.

**Pull request merged**